pkg/run/bootstrap/bootstrap_wizard_ui.go

```diff
@@ -162,62 +194,93 @@ func (m preWizardModel) getContent() string {
 	) + m.textInput.View()
 }
 
-type wizardModel struct {
-	windowIsReady bool
-	viewport      viewport.Model
-	textInputs    []textinput.Model
-	prompts       []string
-	msgChan       chan BootstrapCmdOptions
-	cursorMode    textinput.CursorMode
-	focusIndex    int
-	errorMsg      string
+func makeInput(task *BootstrapWizardTask, isFocused bool) *bootstrapWizardInput {
+	var inputType bootstrapWizardInputType
+
+	if task.isBoolean {
+		inputType = bootstrapWizardInputTypeCheckbox
+	} else {
+		inputType = bootstrapWizardInputTypeTextInput
+	}
+
+	prompt := task.flagName + flagSeparator + task.flagDescription
+
+	ti := textinput.Model{}
+
+	var cb *checkbox
+
+	if inputType == bootstrapWizardInputTypeCheckbox {
+		cb = &checkbox{
+			checked: task.flagValue == "true",
+		}
+	} else {
+		ti = textinput.New()
+		ti.CursorStyle = cursorStyle
+		ti.CharLimit = 100
+
+		ti.SetValue(task.flagValue)
+		ti.Placeholder = task.flagDescription
+
+		if task.isPassword {
+			ti.EchoMode = textinput.EchoPassword
+		}
+
+		if isFocused {
+			ti.Focus()
+			ti.PromptStyle = focusedStyle
+			ti.TextStyle = focusedStyle
+		}
+	}
+
+	return &bootstrapWizardInput{
+		inputType:     inputType,
+		prompt:        prompt,
+		textInput:     ti,
+		checkboxInput: cb,
+	}
 }
 
-func makeTextInput(task *BootstrapWizardTask, isFocused bool) textinput.Model {
-	ti := textinput.New()
-	ti.CursorStyle = cursorStyle
-	ti.CharLimit = 100
+func (input *bootstrapWizardInput) getView(isFocused bool) string {
+	if input.inputType == bootstrapWizardInputTypeTextInput {
+		return input.textInput.View()
+	}
 
-	ti.SetValue(task.flagValue)
-	ti.Placeholder = task.flagDescription
+	var checkmark string
 
-	if task.isPassword {
-		ti.EchoMode = textinput.EchoPassword
+	if input.checkboxInput.checked {
+		checkmark = "x"
+
+		if isFocused {
+			checkmark = focusedStyle.Render(checkmark)
+		}
+	} else {
+		checkmark = blurredStyle.Render("_")
 	}
 
+	open := "["
+	close := "]"
+
 	if isFocused {
-		ti.Focus()
-		ti.PromptStyle = focusedStyle
-		ti.TextStyle = focusedStyle
+		open = focusedStyle.Render(open)
+		close = focusedStyle.Render(close)
 	}
 
-	return ti
+	return fmt.Sprintf("%s%s%s", open, checkmark, close)
```

Comment:
:warning: Could we maybe add the placeholder value here? When I see a line just saying `[x]` I panic that I don't know what it's asking me to do and refuse to read the help text above until I've figured it out, whereas `[x] private` would calm me down enough to pay attention.

Reply:
Added placeholders.